### ISSUE TITLE
Validate against VkPhysicalDeviceLimits

### ISSUE THREAD

**Issue by Jasper-Bekkers** [KhronosGroup/Vulkan-LoaderAndValidationLayers#2063]
Please add validation for a bunch of these since some devices have very low limits and it looks like currently none of these are checked.

```
    uint32_t              maxBoundDescriptorSets;
    uint32_t              maxPerStageDescriptorSamplers;
    uint32_t              maxPerStageDescriptorUniformBuffers;
    uint32_t              maxPerStageDescriptorStorageBuffers;
    uint32_t              maxPerStageDescriptorSampledImages;
    uint32_t              maxPerStageDescriptorStorageImages;
    uint32_t              maxPerStageDescriptorInputAttachments;
    uint32_t              maxPerStageResources;
    uint32_t              maxDescriptorSetSamplers;
    uint32_t              maxDescriptorSetUniformBuffers;
    uint32_t              maxDescriptorSetUniformBuffersDynamic;
    uint32_t              maxDescriptorSetStorageBuffers;
    uint32_t              maxDescriptorSetStorageBuffersDynamic;
    uint32_t              maxDescriptorSetSampledImages;
    uint32_t              maxDescriptorSetStorageImages;
    uint32_t              maxDescriptorSetInputAttachments;
```

**Comment by krOoze** [comment]
I was thinking I would implement this one real quick tomorrow. Or is anybody already working on it?

**Comment by daveh-lunarg** [comment]
Partially resolved by PR #2255. I'll get the rest of them shortly.

**Comment by daveh-lunarg** [comment]
Remaining limits checks implemented in PR #2329
